signer/file/file.go

```diff
@@ -114,7 +157,23 @@ func (fsp FileSignerProvider) Signer(ctx context.Context) (cryptoutil.Signer, er
 	}
 
 	defer keyFile.Close()
-	key, err := cryptoutil.TryParseKeyFromReader(keyFile)
+	// Resolve passphrase: explicit > file > env
+	var pass []byte
+	if fsp.Passphrase != "" {
+		pass = []byte(fsp.Passphrase)
+	} else if fsp.PassphrasePath != "" {
+		b, err := os.ReadFile(fsp.PassphrasePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read key passphrase file: %w", err)
+		}
+		// Trim common line endings
```

Comment:
I guess this is only needed for passphrases stored in a file...  Do you want to error, etc. if they have things after this that you've trimmed instead of silently discarding it?